sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageWithTransactionAsyncSample.java

```diff
@@ -0,0 +1,108 @@
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class ReceiveMessageWithTransactionAsyncSample {
+
+    /**
+     * Main method to invoke this demo on how to use transaction API and settle the {@link ServiceBusReceivedMessage}
+     * as part of a transaction.
+     *
+     * @param args Unused arguments to the program.
+     * @throws InterruptedException If the program is unable to sleep while waiting for the operations to complete.
+     */
+    public static void main(String[] args) throws InterruptedException {
+
+        // The connection string value can be obtained by:
+        // 1. Going to your Service Bus namespace in Azure Portal.
+        // 2. Go to "Shared access policies"
+        // 3. Copy the connection string for the "RootManageSharedAccessKey" policy.
+        String connectionString = "Endpoint={fully-qualified-namespace};SharedAccessKeyName={policy-name};"
+            + "SharedAccessKey={key}";
+
+        // Create a receiver.
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<queue-name>>" will be the name of the Service Bus queue instance you created
+        // inside the Service Bus namespace.
+        ServiceBusClientBuilder builder = new ServiceBusClientBuilder()
+            .connectionString(connectionString);
+
+        // At most, the receiver will automatically renew the message lock until 120 seconds have elapsed.
+        // By default, after messages are processed, the onus is on users to complete, abandon, defer, or dead-letter the
+        // message when they are finished with it.
+
+        ServiceBusReceiverAsyncClient receiver = builder
+            .receiver()
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .queueName("<<queue-name>>")
+            .maxAutoLockRenewalDuration(Duration.ofSeconds(120))
+            .buildAsyncClient();
+
+        ServiceBusSenderAsyncClient sender = builder.sender()
+            .queueName("<<queue-name>>")
+            .buildAsyncClient();
+
+        List<ServiceBusMessage> messages = new ArrayList<>();
+        messages.add(new ServiceBusMessage("Hello world!".getBytes(UTF_8)));
+        messages.add(new ServiceBusMessage("Hello world!".getBytes(UTF_8)));
+
+        // Send some messages.
+        sender.send(messages).block();
+
+        // Hold the transaction.
+        AtomicReference<ServiceBusTransactionContext> transaction = new AtomicReference<>();
+
+        // Transaction is actually started in ServiceBus until you perform first operation
+        // (Example receiver.complete(message, transaction)) with it.
+        // Create transaction and start receiving messages
+        Disposable subscription = receiver.createTransaction()
+            .flatMapMany(transactionContext -> {
```

Comment:
The way we've used transaction here is not reactive. I don't think we want people to have to create an atomic reference every time they create a transaction.